test/rules/no-named-as-default.spec.ts

```diff
@@ -6,17 +6,21 @@ import rule from 'eslint-plugin-import-x/rules/no-named-as-default'
 
 const ruleTester = new TSESLintRuleTester()
 
-console.log({ babel: require(parsers.BABEL) })
-
 ruleTester.run('no-named-as-default', rule, {
   valid: [
+    // https://github.com/un-ts/eslint-plugin-import-x/issues/123
+    'import klawSync from "klaw-sync";',
+    'import ts from "typescript";',
+    `import React from 'react';`,
+    `import classNames from 'classnames';`,
```

Comment:
@Zamiell @controversial I've added the above test cases per your issue and your comment, but the test passed anyway. Would you mind giving me a more specific case?

Reply:
Hi Sukka, thanks for following up on this. I have a monorepo called IsaacScript, you can clone it here: https://github.com/IsaacScript/isaacscript

When I turn on the rule in the "isaacscript-cli" package of the monorepo, the rule results in some false positives:

1) 

```
D:\Repositories\isaacscript\packages\isaacscript-cli\src\banner.ts
  2:8  warning  Using exported name 'figlet' as identifier for default export  import-x/no-named-as-default
```

This is from this line:

```ts
import figlet from "figlet";
```

This is super dangerous because after you change it to `import * as figlet from "figlet"`, TypeScript actually does not throw any errors, but it introduces a run-time error into the code base:

```
TypeError: figlet.textSync is not a function
    at printBanner (file:///C:/Users/win/Desktop/alt-path/src/banner.ts:7:25)
    at monitor (file:///C:/Users/win/Desktop/alt-path/src/commands/monitor/monitor.ts:59:3)
    at Command.<anonymous> (file:///C:/Users/win/Desktop/alt-path/src/commands/monitor/monitor.ts:49:11)
    at Command.listener [as _actionHandler] (C:\Users\win\Desktop\alt-path\node_modules\commander\lib\command.js:542:17)
    at C:\Users\win\Desktop\alt-path\node_modules\commander\lib\command.js:1502:14
    at Command._chainOrCall (C:\Users\win\Desktop\alt-path\node_modules\commander\lib\command.js:1386:12)
    at Command._parseCommand (C:\Users\win\Desktop\alt-path\node_modules\commander\lib\command.js:1501:27)
    at C:\Users\win\Desktop\alt-path\node_modules\commander\lib\command.js:1265:27
    at Command._chainOrCall (C:\Users\win\Desktop\alt-path\node_modules\commander\lib\command.js:1386:12)
    at Command._dispatchSubcommand (C:\Users\win\Desktop\alt-path\node_modules\commander\lib\command.js:1261:25)
```

So in this case the rule is actively destructive.

2)

```
D:\Repositories\isaacscript\packages\isaacscript-cli\src\commands\check\check.ts
  14:8  warning  Using exported name 'klawSync' as identifier for default export  import-x/no-named-as-default
```

This is from this line:

```ts
import klawSync from "klaw-sync";
```

Changing it to `import * as klawSync from "klaw-sync";` causes the file to no longer type-check properly, so the lint rule is a false positive.

3)

```
D:\Repositories\isaacscript\packages\isaacscript-cli\src\prompt.ts
  7:8  warning  Using exported name 'prompt' as identifier for default export  import-x/no-named-as-default
```

This is from this line:

```ts
import prompt from "prompt";
```

Changing it to `import * as prompt from "prompt";` causes the file to no longer type-check properly, so the lint rule is a false positive.

4)

```
D:\Repositories\isaacscript\packages\isaacscript-cli\src\utils.ts
  1:8  warning  Using exported name 'moment' as identifier for default export  import-x/no-named-as-default
```

This is from the line:

```ts
import moment from "moment";
```

Changing it to `import * as moment from "moment";` causes the file to no longer type-check properly, so the lint rule is a false positive.

## Steps to Reproduce

```ts
git clone git@github.com:IsaacScript/isaacscript.git
cd isaacscript
git checkout bug
yarn
cd packages/isaacscript-cli
npx eslint src
```

(change line 62 of "base-import-x.js")

Reply:
The rule `no-named-as-default` is to prevent cases like this:

```ts
import z from 'zod' // bad
import { z } from 'zod' // good
```

In short, `no-named-as-default` prefers named import over default import. It doesn't encourage you to use namespace import.

`figlet`, `prompt`, and `klaw-sync` only have a default export, so the rule should not report any errors. As you can see I've added `klaw-sync` to the valid test cases.

I don't know if I have changed anything in the recent release, but it seems that I can't reproduce those at least in the unit test cases. I will dig further, thanks for your reproduction!

Reply:
@SukkaW I get a `no-named-as-default` error for each of the following:

```ts
import React from 'react';
import z from 'zod';
import classNames from 'classnames';
import value from './my-mod';
```


`my-mod.ts`:
```ts
export const value = 3;
export default value;
```
`node_modules/classnames/index.d.ts`:
```ts
export as namespace classNames;
export = classNames;
```
`node_modules/react/index.d.ts`:
```ts
export = React;
export as namespace React;
```
`node_modules/zod/lib/index.d.ts`:
```ts
import * as z from "./external";
export * from "./external";
export { z };
export default z;
```

---

`eslint.config.js`:
```ts
export default [
  /* ... */
  {
    files: ['**/*.ts', '**/*.tsx', '**/*.d.ts'],
    settings: {
      'import-x/extensions': ['.js', '.jsx', '.mjs', '.cjs', '.ts', '.tsx'],
      'import-x/resolver': { typescript: true },
    },
  },
  /* ... */
]
```

Reply:
@Zamiell I can reproduce your issue in unit tests now! See [`d176da9` (#133)](https://github.com/un-ts/eslint-plugin-import-x/pull/133/commits/d176da904be00967a469a4632b6d2750799a58c2). I am working on fixing this.